packages/client/ui/react-ui/src/providers/CrossmintWalletProvider.tsx

```diff
@@ -0,0 +1,83 @@
+import { useCrossmint } from "@/hooks";
+import { ReactNode, createContext, useEffect, useMemo, useState } from "react";
+
+import { EVMSmartWallet, SmartWalletError, SmartWalletSDK } from "@crossmint/client-sdk-smart-wallet";
+import { Crossmint } from "@crossmint/common-sdk-base";
+
+export type CrossmintWalletConfig = {
+    type: "evm-smart-wallet";
+    defaultChain: "polygon-amoy" | "base-sepolia" | "optimism-sepolia" | "arbitrum-sepolia";
+    createOnLogin: "all-users" | "off";
+};
+
+type WalletStatus = "not-loaded" | "in-progress" | "loaded" | "loading-error";
+
+type ValidWalletState =
+    | { status: "not-loaded" | "in-progress" }
+    | { status: "loaded"; wallet: EVMSmartWallet }
+    | { status: "loading-error"; error: SmartWalletError };
+
+function deriveErrorState(error: unknown): { status: "loading-error"; error: SmartWalletError } {
+    if (error instanceof SmartWalletError) {
+        return { status: "loading-error", error };
+    }
+
+    const message = error instanceof Error ? error.message : String(error);
+    const stack = error instanceof Error ? error.stack : undefined;
+    return { status: "loading-error", error: new SmartWalletError(`Unknown Wallet Error: ${message}`, stack) };
+}
+
+type WalletContext = {
+    status: WalletStatus;
+    wallet?: EVMSmartWallet;
+    error?: SmartWalletError;
+    getOrCreateWallet: () => Promise<void>;
+};
```

Comment:
Exposing this rather than the type intersection defined in the PRD, in practice working with the type intersection sucked. Other libraries (wagmi, tanstack query, nextauth, etc), return similar data w/ their hooks and don't use intersections, likely for similar reasons.

We'll continue to use the type intersection internally so ts enforces only valid states though.